software/control_sw/src/cosmic_fengine.py

```diff
@@ -879,14 +915,177 @@ def set_delay_tracking(self, delays, delay_rates, phases, phase_rates, load_time
+        :param delays: 4-tuple of delays for X and Y polarizations for both tunings. Each value is 
+            the delay, in nanoseconds, which should be applied at the appropriate time.
+            Whole ADC sample delays are implemented using a coarse delay, while sub-sample
+            delays are implemented as a post-FFT phase rotation.
+        :type delays: float
+        :param delay_rates: 4-tuple of delay rates for X and Y polarizations for both tunings. Each value is
+            the delay rate, in nanoseconds per second. The incremental delay
+            which should be added to the current delay each second.
+            Internally, delay rate is converted from nanoseconds-per-second to
+            samples-per-spectra. Firmware delays are updated every 4 spectra.
+        :type delay_rates: float
+        :param phases: 4-tuple of phases for X and Y polarizations for both tunings. Each value is the phase, in radians,
+            which should be applied at the appropriate time.
+        :type phases: float
+        :param phase_rates: 4-tuple of phase rates for X and Y polarizations for both tunings. Each value is the
+            phase rate, in radians per second. This is the incremental phase which should be added
+            to the current phase every second.
+            Internally, phase rate is converted from radians-per-second to radians-per-spectra. 
+            Firmware phases are updated every 4 spectra.
+        :type phase_rates: float
+        :param load_time: Unix time at which delay should be applied. If None, delays are 
+            force-loaded (testing purposes).
+        :type load_time: float
+        :param clock_rate_hz: ADC clock rate in Hz. If None, the clock rate will be computed from
+            the observed PPS interval, which could fail if the PPS is unstable or not present.
+        :type clock_rate_hz: int
+        :param invert_band: If True, invert the gradient of the phase-vs-frequency channel. I.e.,
+            apply a fractional delay which is the negative of the physical delay.
+        :type invert_band: bool
+        """
+        force_delay_load = False
+        if load_time is None:
+            force_delay_load = True
+        else:
+            load_time=int(load_time)
+
+        self.logger.debug("Setting delays at time %s" % (time.ctime(load_time)))
+        self.set_delays(delays, delay_rates, phases, phase_rates,clock_rate_hz,invert_band)
+       
+        #Handle the loading time/force loading of the delays
         if force_delay_load:
             self.delay.force_load()
             self.phaserotate.force_load()
         else:
             print("Loading delays...")
             assert load_time > time.time(), f"""Cannot set a load time that is in the past"""   
-            self.delay.set_target_load_time(load_time)
-            self.phaserotate.set_target_load_time(load_time)
+            self.delay.set_target_load_time(int(load_time*256e6))
```

Comment:
256e6 should probably be parameterised somewhere as `FPGA_CLOCKS_PER_SECOND` or something similar

Reply:
It exists as [FPGA_CLOCK_RATE_HZ](https://github.com/realtimeradio/vla-dev/blob/master/software/control_sw/src/cosmic_fengine.py#L37). I Shall update code accordingly.

Reply:
Fixed in this [commit](https://github.com/realtimeradio/vla-dev/commit/317200ba60bf8cc1fa18ea568de6ffcb0d52c7fb)